src/utils.rs

```diff
@@ -29,7 +35,11 @@ impl NibbleKey {
         NibbleKey(self.0[prefix_length + 1..].to_vec())
     }
 
-    pub fn keep_suffix(&self, suffix_length: usize) -> Self {
+    pub fn prefix(&self, prefix_length: usize) -> Self {
```

Comment:
Do you think a `slice` method which takes a `Range` as input could replace `remove_prefix`, `prefix` and `suffix`?

Reply:
I'm going to try that indeed